ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/WrappingRuleTest.kt

```diff
@@ -1871,6 +1871,303 @@ internal class WrappingRuleTest {
             .hasLintViolation(2, 10, "A newline was expected before 'Bar'")
             .isFormattedAs(formattedCode)
     }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated var initialisation then wrap each expression to a new line`() {
+        val code =
+            """
+            fun f1() {
+                val a = 3; val b = 2
+                val key: String = ""; val lambda: () -> Unit = {  }
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            fun f1() {
+                val a = 3;
+                val b = 2
+                val key: String = "";
+                val lambda: () -> Unit = {  }
+            }
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(2, 15, "Missing newline after \";\""),
+                LintViolation(3, 26, "Missing newline after \";\""),
+            )
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated statements then wrap each expression to a new line`() {
+        val code =
+            """
```

Comment:
Please split in multiple tests. If needed use a nested test class to group related tests.

Reply:
Split the TCs into smaller ones grouped via `Nested` classes